go/vt/vtgate/planbuilder/testdata/cte_cases.json

```diff
@@ -379,25 +379,42 @@
       "QueryType": "SELECT",
       "Original": "with s as (select id from user having count(*) = 1) select * from s",
       "Instructions": {
-        "OperatorType": "Filter",
-        "Predicate": "count(*) = 1",
-        "ResultColumns": 1,
+        "OperatorType": "SimpleProjection",
+        "ColumnNames": [
+          "0:id"
+        ],
+        "Columns": "0",
         "Inputs": [
           {
-            "OperatorType": "Aggregate",
-            "Variant": "Scalar",
-            "Aggregates": "any_value(0) AS id, sum_count_star(1) AS count(*)",
+            "OperatorType": "Projection",
+            "Expressions": [
+              ":0 as id",
+              "count(*) = 1 as count(*) = 1"
+            ],
             "Inputs": [
```

Comment:
SimpleProjection on top of Projection 🤔

Reply:
@systay Is this what is causing the multi nested derived table query on v19? 🤔